src/input/mod.rs

```diff
@@ -969,20 +1001,25 @@ impl State {
                                     .set_overview_mode(None, self.common.event_loop_handle.clone());
                             }
                         }
+
                         std::mem::drop(shell);
                     };
 
                     let ptr = seat.get_pointer().unwrap();
-                    ptr.button(
-                        self,
-                        &ButtonEvent {
-                            button,
-                            state: event.state(),
-                            serial,
-                            time: event.time_msec(),
-                        },
-                    );
-                    ptr.frame(self);
+                    if pass_event {
+                        ptr.button(
+                            self,
+                            &ButtonEvent {
+                                button,
+                                state: event.state(),
+                                serial,
+                                time: event.time_msec(),
+                            },
+                        );
+                        ptr.frame(self);
+                    } else {
```

Comment:
Small nitpick here: Given we set the grab with an idle-callback unsetting any grab any time we don't pass events isn't wrong, but technically we only want to unset grabs on a `Released`-event.

```suggestion
                    } else if event.state() == ButtonState::Released {
```